### PR TITLE
feat: add expo config plugin

### PR DESCRIPTION
**Pull request by intent-kacper-cyranowski**
Added code from [config plugin](https://github.com/expo/config-plugins/tree/main/packages/react-native-ble-plx) to support expo.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,6 +37,7 @@ It does NOT support:
 3. [Documentation & Support](#documentation--support)
 4. [Configuration & Installation](#configuration--installation)
 5. [Troubleshooting](#troubleshooting)
+6. [Contributions](#contributions)
```
Comment (dominik-czupryna-withintent) [inline README.md:40]: if looks like more thanks sections 😛

File: plugin/src/__tests__/withBLEAndroidManifest-test.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      }
+    })
+    expect(androidManifest.manifest['uses-permission-sdk-23']).toContainEqual({
+      $: {
+        'android:name': 'android.permission.ACCESS_FINE_LOCATION'
+      }
+    })
+    // Sanity
+    expect(XML.format(androidManifest)).toMatch(
+      /<uses-permission-sdk-23 android:name="android\.permission\.ACCESS_COARSE_LOCATION"\/>/
```
Comment (dominik-czupryna-withintent) [inline plugin/src/__tests__/withBLEAndroidManifest-test.ts:30]: ACCESS_COARSE_LOCATION is redundant when you have ACCESS_FINE_LOCATION

File: plugin/src/withBLE.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  // iOS
+  config = withBluetoothPermissions(config, _props)
+  config = withBLEBackgroundModes(config, _props.modes || [])
+
+  // Android
+  config = AndroidConfig.Permissions.withPermissions(config, [
+    'android.permission.BLUETOOTH',
+    'android.permission.BLUETOOTH_ADMIN',
+    'android.permission.BLUETOOTH_CONNECT' // since Android SDK 31
```
Comment (dominik-czupryna-withintent) [inline plugin/src/withBLE.ts:39]: what about BLUETOOTH_SCAN (https://developer.android.com/reference/android/Manifest.permission#BLUETOOTH_SCAN)

**Review by dominik-czupryna-withintent — APPROVED** [review]

File: plugin/src/withBLE.ts (hunk 3 above)
Comment (intent-kacper-cyranowski) [reply plugin/src/withBLE.ts:39]: `BLUETOOTH_SCAN` is dynamically added in `addScanPermissionToManifest`

**Review by dominik-czupryna-withintent — APPROVED** [review]

**Pull request merged**